crates/notedeck_columns/src/contacts.rs

```diff
@@ -0,0 +1,62 @@
+use nostrdb::{Filter, Ndb, Note, Transaction};
+
+fn pk1_is_following_pk2(
+    ndb: &Ndb,
+    txn: &Transaction,
+    pk1: &[u8; 32],
+    pk2: &[u8; 32],
+) -> Option<bool> {
+    let note = get_contacts_note(ndb, txn, pk1)?;
+
+    Some(note_follows(note, pk2))
+}
+
+pub fn trust_media_from_pk2(
+    ndb: &Ndb,
+    txn: &Transaction,
+    pk1: Option<&[u8; 32]>,
+    pk2: &[u8; 32],
+) -> bool {
+    pk1.map(|pk| pk1_is_following_pk2(ndb, txn, pk, pk2).unwrap_or(false))
+        .unwrap_or(false)
+}
+
+fn get_contacts_note<'a>(ndb: &'a Ndb, txn: &'a Transaction, user: &[u8; 32]) -> Option<Note<'a>> {
+    Some(
+        ndb.query(txn, &[contacts_filter(user)], 1)
+            .ok()?
+            .first()?
+            .note
+            .clone(),
+    )
+}
+
+pub fn contacts_filter(pk: &[u8; 32]) -> Filter {
+    Filter::new().authors([pk]).kinds([3]).limit(1).build()
+}
+
+fn note_follows(contacts_note: Note<'_>, pk: &[u8; 32]) -> bool {
```

Comment:
it's funny, normally I would have said this is too slow but these are all non-copy iterators over compact binary data. so it's likely fast enough until we have the follow cache.